### PR TITLE
Core: move `OptionGroup` definition to `Options.py`

### PR DESCRIPTION
**Pull request by beauxq**
## What is this fixing or adding?

`BaseClasses.py` doesn't seem like the right place for this.
- `BaseClasses.py` was otherwise untouched by the PR that introduced this feature
- World devs have already noticed that it's nicer to define their `OptionGroup`s in their world's options module, which often doesn't import anything else from `BaseClasses.py`
  - Adding `BaseClasses` to those imports increases the risk of having to deal with circular import issues.

## How was this tested?

opened an option page from WebHost, observed the option groups, generated a game from the option page

### CODE REVIEW

File: Options.py [hunk 1]
```diff
@@ -1123,6 +1123,14 @@ class DeathLinkMixin:
     death_link: DeathLink
 
 
+class OptionGroup(typing.NamedTuple):
+    """Define a grouping of options"""
+    name: str
+    """Name of the group to categorize this option in for display on the WebHost and in generated YAMLS."""
+    options: typing.List[typing.Type[Option[typing.Any]]]
+    """Options to be in the defined group. """
```
Comment (alwaysintreble) [inline Options.py:1131]:
```suggestion
    """Options to be in the defined group."""
```

File: Options.py [hunk 2]
```diff
@@ -1123,6 +1123,14 @@ class DeathLinkMixin:
     death_link: DeathLink
 
 
+class OptionGroup(typing.NamedTuple):
+    """Define a grouping of options"""
+    name: str
+    """Name of the group to categorize this option in for display on the WebHost and in generated YAMLS."""
```
Comment (alwaysintreble) [inline Options.py:1129]:
```suggestion
    """Name of the group to categorize these options in for display on the WebHost and in generated YAMLS."""
```

File: Options.py [hunk 3]
```diff
@@ -1123,6 +1123,14 @@ class DeathLinkMixin:
     death_link: DeathLink
 
 
+class OptionGroup(typing.NamedTuple):
+    """Define a grouping of options"""
```
Comment (alwaysintreble) [inline Options.py:1127]:
```suggestion
    """Define a grouping of options."""
```

**Review by alwaysintreble — CHANGES_REQUESTED** [review]

**Review by Exempt-Medic — APPROVED** [review]
Change makes sense and fixes what I thought was an unusual decision. Tested it locally, it worked.

**Review by alwaysintreble — APPROVED** [review]

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**